keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/ScalarDoubleTensor.java

```diff
@@ -637,6 +645,17 @@ public DoubleTensor standardizeInPlace() {
         throw new IllegalStateException("Cannot standardize a scalar");
     }
 
+    @Override
+    public DoubleTensor zero() {
```

Comment:
This could be the default method in the `DoubleTensor` interface

Reply:
I'm not sure what you mean by this can you explain?

Reply:
Instead of having an identical `zero` method in both `ScalarDoubleTensor` and `Nd4jDoubleTensor`, you can put it in `DoubleTensor` and use the `default` keyword (introduced in Java 8).